### PR TITLE
Adding Bicep template for App Configuration with Key Vault reference

### PR DESCRIPTION
**Pull request by willvelida**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@ZhijunZhao - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@ZhijunZhao - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@ZhijunZhao - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by willvelida** [comment]
Hey @ZhijunZhao, not sure who to contact to, but the build fails due to a timeout occurring when the template is being deployed. The build logs show that the template is valid, so should the timeout period be extended or should we reattempt the build? (I don't have permissions to do this)

**Comment by azure-quickstarts** [comment]
@ZhijunZhao - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by willvelida** [comment]
@bmoore-msft Please review :)

File: quickstarts/microsoft.appconfiguration/app-configuration-store-keyvaultref/main.bicep [hunk 1]
```diff
@@ -0,0 +1,32 @@
+@description('Specifies the name of the App Configuration store.')
+param configStoreName string = 'appconfig${uniqueString(resourceGroup().id)}'
+
+@description('Specifies the Azure location where the app configuration store should be created.')
+param location string = resourceGroup().location
+
+@description('Specifies the name of the key-value resource. The name is a combination of key and label with $ as delimiter.')
+param keyValue string = 'KeyVaultReferenceSample'
```
[the remark this answers is not in the document]
Comment (willvelida) [reply quickstarts/microsoft.appconfiguration/app-configuration-store-keyvaultref/main.bicep]: Fixed in latest commit

**Comment by azure-quickstarts** [comment]
@ZhijunZhao - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**